# Cleaning CIC IoT-DIAD Dataset

## Overview: Loading, Initial Observations, and Cleaning CIC IoT-DIAD 2024 dataset


---
## Project Workflow:
1. **Step 1: Install and Import Necessary Libraries**
2. **Step 2: Access Data**
3. **Step 3: Check Packaging**
5. **Step 4: Clean Data**



# Step 1: Install and Import Necessary Libraries


In [1]:
import pandas as pd
import numpy as np


# Step 2: Access Data

In [2]:
# brute force
file_path = r"data/DictionaryBruteForce.csv"

BF_data = pd.read_csv(file_path)

In [3]:
file_path = r"data/BenignTraffic.csv"

BN_data = pd.read_csv(file_path)

In [4]:
# mirai

file_path = r"data/Mirai-greip_flood.csv"

Mirai_data = pd.read_csv(file_path)

/var/folders/0_/p6_60b2d3r33yjj108yb0_7r0000gn/T/ipykernel_26366/3044158186.py:5: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  Mirai_data = pd.read_csv(file_path)


In [5]:
# XSS

file_path = r"data/XSS.csv"

XSS_data = pd.read_csv(file_path)

In [6]:
# Command Injection

file_path = r"data/CommandInjection.csv"

CI_data = pd.read_csv(file_path)

/var/folders/0_/p6_60b2d3r33yjj108yb0_7r0000gn/T/ipykernel_26366/2916856881.py:5: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  CI_data = pd.read_csv(file_path)


In [7]:
# Backdoor Malware

file_path = r"data/Backdoor_Malware.csv"

BM_data = pd.read_csv(file_path)

# Step 3: Check Packaging

## Brute Force

In [8]:
print(BF_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131477 entries, 0 to 131476
Columns: 135 entries, stream to l3_ip_dst_count
dtypes: float64(97), int64(22), object(16)
memory usage: 135.4+ MB
None


In [9]:
print(BF_data.head())


   stream             src_mac            dst_mac           src_ip  \
0       0  Nest Indoor Camera  3c:18:a0:41:c3:a0  192.168.137.208   
1       0    Yi Indoor Camera  3c:18:a0:41:c3:a0   192.168.137.52   
2       1    Yi Indoor Camera  3c:18:a0:41:c3:a0   192.168.137.52   
3       2    Yi Indoor Camera  3c:18:a0:41:c3:a0   192.168.137.52   
4       1   94:39:e5:5d:27:a6  3c:18:a0:41:c3:a0  192.168.137.174   

            dst_ip  src_port  dst_port  inter_arrival_time  \
0    35.186.43.132     53345       443            0.000000   
1    47.254.89.110     10124     32100            0.052544   
2    47.90.209.117     10124     32100            0.052548   
3  161.117.254.158     10124     32100            0.052549   
4   185.209.178.72      4608       443            0.212556   

   time_since_previously_displayed_frame  port_class_dst  ...   sum_p  min_p  \
0                               0.000000               1  ...  2040.0  102.0   
1                               0.052544            

## Benign

In [10]:
print(BN_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 297063 entries, 0 to 297062
Columns: 135 entries, stream to l3_ip_dst_count
dtypes: float64(97), int64(22), object(16)
memory usage: 306.0+ MB
None


In [11]:
print(BN_data.head())

   stream               src_mac            dst_mac           src_ip  \
0       0  Arlo Q Indoor Camera  3c:18:a0:41:c3:a0  192.168.137.175   
1       0  Arlo Q Indoor Camera  3c:18:a0:41:c3:a0  192.168.137.175   
2       0  Arlo Q Indoor Camera  3c:18:a0:41:c3:a0  192.168.137.175   
3       0  Arlo Q Indoor Camera  3c:18:a0:41:c3:a0  192.168.137.175   
4       0  Arlo Q Indoor Camera  3c:18:a0:41:c3:a0  192.168.137.175   

         dst_ip  src_port  dst_port  inter_arrival_time  \
0  99.81.244.93     56891       443            0.000000   
1  99.81.244.93     56891       443            0.000164   
2  99.81.244.93     56891       443            0.000269   
3  99.81.244.93     56891       443            0.000414   
4  99.81.244.93     56891       443            0.001800   

   time_since_previously_displayed_frame  port_class_dst  ...    sum_p  \
0                               0.000000               1  ...  58160.0   
1                               0.000164               1  ...  58160.0

## Mirai

In [12]:
Mirai_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 213030 entries, 0 to 213029
Columns: 135 entries, stream to l3_ip_dst_count
dtypes: float64(97), int64(22), object(16)
memory usage: 219.4+ MB


In [13]:
Mirai_data.head()

stream             src_mac             dst_mac           src_ip  \
0       0     SmartThings Hub   3c:18:a0:41:c3:a0    192.168.137.3   
1       1  Nest Indoor Camera   3c:18:a0:41:c3:a0   192.168.137.91   
2       1   3c:18:a0:41:c3:a0  Nest Indoor Camera    35.185.44.250   
3      -1   24:05:88:30:6f:89   3c:18:a0:41:c3:a0  192.168.137.100   
4       2       Teckin Plug 1   3c:18:a0:41:c3:a0  192.168.137.171   

           dst_ip  src_port  dst_port  inter_arrival_time  \
0    52.15.60.193     52190       443            0.000000   
1   35.185.44.250     40159       443            0.196559   
2  192.168.137.91       443     40159            0.245207   
3   192.168.137.1         0         0            0.350814   
4   54.187.112.38     49424       443            0.372572   

   time_since_previously_displayed_frame  port_class_dst  ...   sum_p  min_p  \
0                               0.000000               1  ...  9160.0  458.0   
1                               0.196559               1  ...  2040.0  102.0   
2                               0.048648               2  ...  1950.0   12.0   
3                               0.105607               0  ...     0.0    0.0   
4                               0.021758               1  ...  4580.0  229.0   

   max_p  med_p  average_p  var_p   q3_p   q1_p  iqr_p  l3_ip_dst_count  
0  458.0  458.0      458.0    0.0  458.0  458.0    0.0              1.0  
1  102.0  102.0      102.0    0.0  102.0  102.0    0.0              1.0  
2  102.0  102.0       97.5  405.0  102.0  102.0    0.0              2.0  
3    0.0    0.0        0.0    0.0    0.0    0.0    0.0              1.0  
4  229.0  229.0      229.0    0.0  229.0  229.0    0.0              1.0  

[5 rows x 135 columns]

## XSS

In [14]:
XSS_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40101 entries, 0 to 40100
Columns: 135 entries, stream to l3_ip_dst_count
dtypes: float64(97), int64(22), object(16)
memory usage: 41.3+ MB


In [15]:
XSS_data.head()

stream            src_mac                                     dst_mac  \
0       0  3c:18:a0:41:c3:a0                     Borun/Sichuan-AI Camera   
1       0  3c:18:a0:41:c3:a0                            Amazon Echo Show   
2       1  3c:18:a0:41:c3:a0                     Borun/Sichuan-AI Camera   
3      -1  3c:18:a0:41:c3:a0  Wemo smart plug 1 (Wemo id: Wemo.Mini.AD3)   
4       1  3c:18:a0:41:c3:a0                          Nest Indoor Camera   

           src_ip           dst_ip  src_port  dst_port  inter_arrival_time  \
0    47.254.83.84  192.168.137.163      7788     59336            0.000000   
1  34.158.253.218  192.168.137.228      4070     48752            0.035363   
2   192.168.137.1  192.168.137.163        53     44205            0.148731   
3             NaN              NaN         0         0            0.204984   
4   35.185.44.250   192.168.137.34       443     54482            0.277251   

   time_since_previously_displayed_frame  port_class_dst  ...   sum_p  min_p  \
0                               0.000000               3  ...   360.0   18.0   
1                               0.035363               2  ...   365.0   18.0   
2                               0.113368               2  ...  1060.0   53.0   
3                               0.056253               2  ...   347.0    0.0   
4                               0.072267               3  ...   240.0   12.0   

   max_p  med_p  average_p      var_p  q3_p  q1_p  iqr_p  l3_ip_dst_count  
0   18.0   18.0      18.00   0.000000  18.0  18.0    0.0              1.0  
1   23.0   18.0      18.25   1.250000  18.0  18.0    0.0              2.0  
2   53.0   53.0      53.00   0.000000  53.0  53.0    0.0              1.0  
3   23.0   18.0      17.35  17.923684  18.0  18.0    0.0              3.0  
4   12.0   12.0      12.00   0.000000  12.0  12.0    0.0              1.0  

[5 rows x 135 columns]

## Command Injection

In [16]:
CI_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55632 entries, 0 to 55631
Columns: 135 entries, stream to l3_ip_dst_count
dtypes: float64(97), int64(22), object(16)
memory usage: 57.3+ MB


In [17]:
CI_data.head()

stream            src_mac                  dst_mac        src_ip  \
0      -1  e8:1b:69:f8:d4:e3  Borun/Sichuan-AI Camera           NaN   
1       0  3c:18:a0:41:c3:a0  Borun/Sichuan-AI Camera  47.254.83.84   
2       0  3c:18:a0:41:c3:a0  Borun/Sichuan-AI Camera  47.254.83.84   
3       0  3c:18:a0:41:c3:a0  Borun/Sichuan-AI Camera  47.254.83.84   
4       0  3c:18:a0:41:c3:a0  Borun/Sichuan-AI Camera  47.254.83.84   

            dst_ip  src_port  dst_port  inter_arrival_time  \
0              NaN         0         0            0.000000   
1  192.168.137.163      1340     52435            0.261533   
2  192.168.137.163      1340     52435            0.350276   
3  192.168.137.163      1340     52435            0.350647   
4  192.168.137.163      9001     39676            0.380779   

   time_since_previously_displayed_frame  port_class_dst  ...  sum_p  min_p  \
0                               0.000000               2  ...    0.0    0.0   
1                               0.261533               3  ...   20.0    0.0   
2                               0.088743               3  ...   32.0    0.0   
3                               0.000371               3  ...   44.0    0.0   
4                               0.030132               2  ...   62.0    0.0   

   max_p  med_p  average_p      var_p q3_p  q1_p  iqr_p  l3_ip_dst_count  
0    0.0    0.0        0.0   0.000000  0.0   0.0    0.0              1.0  
1   20.0    0.0        1.0  20.000000  0.0   0.0    0.0              2.0  
2   20.0    0.0        1.6  25.936842  0.0   0.0    0.0              2.0  
3   20.0    0.0        2.2  31.115789  0.0   0.0    0.0              2.0  
4   20.0    0.0        3.1  43.147368  0.0   0.0    0.0              2.0  

[5 rows x 135 columns]

## Backdoor Malware

In [18]:
BM_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33349 entries, 0 to 33348
Columns: 135 entries, stream to l3_ip_dst_count
dtypes: float64(97), int64(22), object(16)
memory usage: 34.3+ MB


In [19]:
BM_data.head()

stream            src_mac             dst_mac         src_ip  \
0       0  3c:18:a0:41:c3:a0         Wyze Camera   52.32.107.17   
1      -1  dc:a6:32:dc:27:d5  Philips Hue Bridge            NaN   
2      -1  00:a3:d1:07:6f:03   00:a3:d1:07:6f:03            NaN   
3       1  3c:18:a0:41:c3:a0  Nest Indoor Camera  35.185.44.250   
4      -1  00:a3:d1:07:6f:03   01:80:c2:00:00:00            NaN   

            dst_ip  src_port  dst_port  inter_arrival_time  \
0  192.168.137.179       443     48531            0.000000   
1              NaN         0         0            0.102862   
2              NaN         0         0            0.187258   
3   192.168.137.34       443     54482            0.189870   
4              NaN         0         0            0.396113   

   time_since_previously_displayed_frame  port_class_dst  ...  sum_p  min_p  \
0                               0.000000               2  ...  240.0   12.0   
1                               0.102862               2  ...    0.0    0.0   
2                               0.084396               2  ...    0.0    0.0   
3                               0.002612               3  ...  240.0   12.0   
4                               0.206243               3  ...    0.0    0.0   

   max_p  med_p  average_p  var_p  q3_p  q1_p  iqr_p  l3_ip_dst_count  
0   12.0   12.0       12.0    0.0  12.0  12.0    0.0              1.0  
1    0.0    0.0        0.0    0.0   0.0   0.0    0.0              1.0  
2    0.0    0.0        0.0    0.0   0.0   0.0    0.0              1.0  
3   12.0   12.0       12.0    0.0  12.0  12.0    0.0              1.0  
4    0.0    0.0        0.0    0.0   0.0   0.0    0.0              1.0  

[5 rows x 135 columns]

### Results

- keep all communication details
    - `stream`, session unique identifier
    - `user_agent`, flag suspicious devices interactign with cameras
    - **important** `src_mac` seems to be the source device name. this should be kept if I want to focus on a few select cameras or brands 
    - `dst_mac` seems to be the attackers device name. should be kept for analysis
    - `dst_ip`, `dst_port`
- keep `time_since_previously_displayed_frame`: The time since the last similar activity was noticed
- keep `inter_arrival_time` - Sudden changes in timing between packets might suggest an anomaly, like brute force attempts or unauthorized streaming
- data details
    - `eth_size`
    - **important** `payload_length`
            - May highlight abnormal amounts of data being transferred from the camera, like someone streaming the video feed without authorization 
- reliability
    - `l4_tcp` if the data is being sent using a reliable communication method
    - `l4_udp` if the data is being sent using a quicker, less reliable method
- security 
    - `payload_entropy` A way to measure how random or messy the data inside looks—it helps detect suspicious activity
        - High entropy might suggest encrypted or compressed data, which could mean unauthorized parties are exfiltrating sensitive data
    - `user_agent` - May identify unusual or fake devices attempting to interact with cameras
- DNS and HTTPS info
    - `dns_server`, `dns_len_qry`: Look for unusual DNS activity, such as queries to strange domains that don't belong to your system
    - `http_request_method`, `http_host`, `http_uri`: Unusual HTTP requests (e.g., POST when only GET is expected) or access to unauthorized hosts may signal a breach
    - `http_response_code`: Repeated response codes like 401 Unauthorized could indicate unauthorized access attempts
- streaming
    - `stream_jitter`, High jitter could indicate unauthorized streaming or unstable connections
    - `stream_count`, `src_ip_count`, `channel_count`: Spikes in counts could correlate to brute force login attempts, scans, or bulk data transmission


# Step 4: Clean Data

## Drop Unecessary Variables

In [20]:
columns_to_keep = [
    'stream', 'user_agent', 'src_mac', 'dst_mac', 'dst_ip', 'dst_port', 
    'time_since_previously_displayed_frame', 'inter_arrival_time', 'eth_size', 
    'payload_length', 'l4_tcp', 'l4_udp', 'payload_entropy', 'user_agent', 
    'dns_server', 'dns_len_qry', 'http_request_method', 
    'http_host', 'http_uri', 'http_response_code', 'stream_jitter_10_sum',
    'stream_jitter_10_mean', 'stream_jitter_10_var', 'stream_10_count',
    'stream_10_mean', 'stream_10_var',  'src_ip_10_count', 'src_ip_10_mean', 
    'src_ip_10_var', 'channel_10_count', 'channel_10_mean', 'channel_10_var'
]

In [21]:
BF_data = BF_data[columns_to_keep]
BN_data = BN_data[columns_to_keep]
Mirai_data = Mirai_data[columns_to_keep]
XSS_data = XSS_data[columns_to_keep]
CI_data = CI_data[columns_to_keep]
BM_data = BM_data[columns_to_keep]


### drop any NAs

In [22]:
BF_data = BF_data.dropna()
BN_data = BN_data.dropna()
Mirai_data = Mirai_data.dropna()
XSS_data = XSS_data.dropna()
CI_data = CI_data.dropna()
BM_data = BM_data.dropna()


## Focus on Specific Brands/Cameras

In [23]:
# print how many unique devices there are
print(BF_data['src_mac'].nunique())
print(BF_data['src_mac'].unique())

66
['Nest Indoor Camera' 'SmartThings Hub' '94:39:e5:5d:27:a6'
 '3c:18:a0:41:c3:a0' 'HeimVision SmartLife Radio/Lamp'
 'Borun/Sichuan-AI Camera' 'Google Nest Mini Speaker' '56:4f:8a:e1:f3:2d'
 'Amazon Echo Dot 2' 'Amazon Echo Spot' 'LampUX RGB'
 'Gosund Power strip (1)' 'Levoit Air Purifier' 'Sonos One Speaker'
 'Wyze Camera' 'dc:a6:32:dc:27:d5' 'DCS8000LHA1 D-Link Mini Camera'
 'Rbcior Camera' 'GoSund Smart plug WP2 (1)' 'Atomi Coffee Maker'
 'd4:81:d7:97:d0:29' 'SIMCAM 1S (AMPAKTec)' 'Yutron Plug 2'
 'Gosund Smart Plug WP3 (2)' 'AMCREST WiFi Camera'
 'GoSund Smart Plug WP2 (2)' 'Yutron Plug 1' 'GoSund Smart plug WP2 (3)'
 'GoSund Power strip (2)' 'LIFX Lightbulb' 'Govee Smart Humidifer'
 'e8:1b:69:f8:d6:e6' 'Globe Lamp ESP_B1680C' 'Teckin Plug 2'
 'Amazon Echo Studio' 'Amazon Echo Show' 'Amazon Echo Dot 1'
 'Arlo Base Station' 'Teckin Light Strip' 'GoSund Bulb' 'Lumiman bulb'
 'Teckin Plug 1' '24:05:88:30:6f:89' 'AeoTec Smart Home Hub'
 'Netatmo Camera' 'Arlo Q Indoor Camera' 'b0:09:

In [24]:
# make new dfs only contianing camera devices being attacked/accessed
camera_BF_data = BF_data[BF_data['src_mac'].str.contains('camera|cam|show', case=False, na=False)]
camera_BN_data = BN_data[BN_data['src_mac'].str.contains('camera|cam|show', case=False, na=False)]
camera_CI_data = CI_data[CI_data['src_mac'].str.contains('camera|cam|show', case=False, na=False)]
camera_Mirai_data = Mirai_data[Mirai_data['src_mac'].str.contains('camera|cam|show', case=False, na=False)]
camera_XSS_data = XSS_data[XSS_data['src_mac'].str.contains('camera|cam|show', case=False, na=False)]
camera_BM_data = BM_data[BM_data['src_mac'].str.contains('camera|cam|show', case=False, na=False)]

In [25]:
# number of unique cameras
print(camera_BF_data['src_mac'].nunique())
print(camera_BN_data['src_mac'].nunique())
print(camera_CI_data['src_mac'].nunique())
print(camera_Mirai_data['src_mac'].nunique())
print(camera_XSS_data['src_mac'].nunique())
print(camera_BM_data['src_mac'].nunique())



16
14
15
14
16
16


In [26]:
# common devices
common_src_mac = set(camera_BF_data['src_mac']).intersection(
    camera_BN_data['src_mac'],
    camera_CI_data['src_mac'],
    camera_Mirai_data['src_mac'],
    camera_XSS_data['src_mac'],
    camera_BM_data['src_mac']
)

camera_BF_data = camera_BF_data[camera_BF_data['src_mac'].isin(common_src_mac)]
camera_BN_data = camera_BN_data[camera_BN_data['src_mac'].isin(common_src_mac)]
camera_CI_data = camera_CI_data[camera_CI_data['src_mac'].isin(common_src_mac)]
camera_Mirai_data = camera_Mirai_data[camera_Mirai_data['src_mac'].isin(common_src_mac)]
camera_XSS_data = camera_XSS_data[camera_XSS_data['src_mac'].isin(common_src_mac)]
camera_BM_data = camera_BM_data[camera_BM_data['src_mac'].isin(common_src_mac)]


print("Common src_mac values:", common_src_mac)


Common src_mac values: {'DCS8000LHA1 D-Link Mini Camera', 'Nest Indoor Camera', 'TP-Link Tapo Camera', 'Netatmo Camera', 'Home Eye Camera', 'AMCREST WiFi Camera', 'Yi Indoor Camera', 'HeimVision Smart WiFi Camera', 'Rbcior Camera', 'Wyze Camera', 'Arlo Q Indoor Camera', 'Amazon Echo Show'}


In [27]:
# number of unique cameras
print(camera_BF_data['src_mac'].nunique())
print(camera_BN_data['src_mac'].nunique())
print(camera_CI_data['src_mac'].nunique())
print(camera_Mirai_data['src_mac'].nunique())
print(camera_XSS_data['src_mac'].nunique())
print(camera_BM_data['src_mac'].nunique())

12
12
12
12
12
12


In [28]:
print(common_src_mac)

{'DCS8000LHA1 D-Link Mini Camera', 'Nest Indoor Camera', 'TP-Link Tapo Camera', 'Netatmo Camera', 'Home Eye Camera', 'AMCREST WiFi Camera', 'Yi Indoor Camera', 'HeimVision Smart WiFi Camera', 'Rbcior Camera', 'Wyze Camera', 'Arlo Q Indoor Camera', 'Amazon Echo Show'}


## Combine Data

In [29]:
camera_BF_data['traffic_type'] = 'BF'
camera_BN_data['traffic_type'] = 'BN'
camera_CI_data['traffic_type'] = 'CI'
camera_Mirai_data['traffic_type'] = 'Mirai'
camera_XSS_data['traffic_type'] = 'XSS'
camera_BM_data['traffic_type'] = 'BM'


In [30]:
combined_data = pd.concat([camera_BF_data, camera_BN_data, camera_CI_data, camera_Mirai_data, camera_XSS_data, camera_BM_data], ignore_index=True)

print(combined_data.info())
# should be 76908 data points 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 76908 entries, 0 to 76907
Data columns (total 33 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   stream                                 76908 non-null  int64  
 1   user_agent                             76908 non-null  object 
 2   src_mac                                76908 non-null  object 
 3   dst_mac                                76908 non-null  object 
 4   dst_ip                                 76908 non-null  object 
 5   dst_port                               76908 non-null  int64  
 6   time_since_previously_displayed_frame  76908 non-null  float64
 7   inter_arrival_time                     76908 non-null  float64
 8   eth_size                               76908 non-null  int64  
 9   payload_length                         76908 non-null  int64  
 10  l4_tcp                                 76908 non-null  int64  
 11  l4

In [31]:
# numper of instances of each traffic type

traffic_type_counts = combined_data['traffic_type'].value_counts()
print(traffic_type_counts)

traffic_type
BN       35529
BF       20552
CI        6962
XSS       6612
BM        6138
Mirai     1115
Name: count, dtype: int64


## Clean Data Points

In [32]:
combined_data['src_mac'] = combined_data['src_mac'].replace('DCS8000LHA1 D-Link Mini Camera', 'D-Link Mini Camera')

In [33]:
combined_data.sample(10)

stream user_agent           src_mac            dst_mac          dst_ip  \
39500    3716       none     Rbcior Camera  3c:18:a0:41:c3:a0     54.188.99.4   
66983      59       none       Wyze Camera  3c:18:a0:41:c3:a0   66.23.204.162   
47959      27       none  Amazon Echo Show  Amazon Echo Dot 2  192.168.137.58   
30530     978       none  Amazon Echo Show  3c:18:a0:41:c3:a0   3.223.181.245   
62259     467       none    Netatmo Camera  3c:18:a0:41:c3:a0   51.145.143.28   
9309     1530       none    Netatmo Camera  3c:18:a0:41:c3:a0   51.145.143.28   
45660      27       none  Amazon Echo Show  Amazon Echo Dot 2  192.168.137.58   
72460     194       none  Amazon Echo Show  3c:18:a0:41:c3:a0  13.225.189.194   
69281     379       none       Wyze Camera  3c:18:a0:41:c3:a0   52.39.102.254   
4887      898       none  Yi Indoor Camera  3c:18:a0:41:c3:a0   47.254.14.172   

       dst_port  time_since_previously_displayed_frame  inter_arrival_time  \
39500      8883                               0.034431         2370.148130   
66983     10001                               0.000559         1151.998821   
47959     55444                               0.011124         3572.753034   
30530       443                               0.000022          644.279592   
62259       443                               0.006896         2127.670554   
9309        443                               0.004222       249537.112598   
45660     55444                               0.000047         3062.736036   
72460       443                               0.000001         1273.851037   
69281       443                               0.046730         1921.332698   
4887        443                               0.000001       249037.743437   

       eth_size  payload_length  ...  stream_10_count  stream_10_mean  \
39500        66               0  ...             30.0      235.100000   
66983        90              48  ...              2.0       75.000000   
47959        60              16  ...             42.0       60.952381   
30530      1514            1448  ...              8.0     1765.750000   
62259      1494            1428  ...             27.0      883.481481   
9309       1494            1428  ...            153.0      984.954248   
45660        60              16  ...             41.0       60.975610   
72460        66               0  ...             77.0     1340.116883   
69281       192             126  ...             14.0      495.571429   
4887         60               0  ...             14.0      415.142857   

       stream_10_var src_ip_10_count src_ip_10_mean  src_ip_10_var  \
39500   3.096713e+05            17.0     103.941176   3.426934e+03   
66983   4.500000e+02            44.0     153.181818   5.958029e+04   
47959   1.404646e+01            30.0      98.833333   1.906794e+04   
30530   2.413045e+06            14.0    1101.357143   1.968851e+06   
62259   2.040821e+06            15.0    1158.600000   3.248385e+06   
9309    4.909195e+05           507.0    1379.201183   2.855758e+05   
45660   1.437439e+01            27.0      90.296296   1.971845e+04   
72460   1.810536e+05            49.0    1832.122449   1.496649e+07   
69281   3.935275e+05            53.0     150.547170   5.231933e+04   
4887    2.797678e+05            15.0     125.800000   1.717717e+04   

      channel_10_count channel_10_mean channel_10_var  traffic_type  
39500             30.0      235.100000   3.096713e+05            BN  
66983              2.0       75.000000   4.500000e+02           XSS  
47959             49.0      101.326531   1.962360e+04            BN  
30530              8.0     1765.750000   2.413045e+06            BN  
62259             27.0      883.481481   2.040821e+06            CI  
9309             853.0      875.854631   5.970393e+05            BF  
45660             43.0       84.325581   1.321318e+04            BN  
72460             77.0     1340.116883   1.810536e+05            BM  
69281             14.0      495.571429   3.935275e+

In [34]:
combined_data.to_csv('data/CIC_Camera_data.csv', index=False)

## Results


- I will be focusing on the cameras: {'Yi Indoor Camera', 'Nest Indoor Camera', 'AMCREST WiFi Camera', 'Home Eye Camera', 'Rbcior Camera', 'DCS8000LHA1 D-Link Mini Camera', 'Netatmo Camera', 'Arlo Q Indoor Camera', 'TP-Link Tapo Camera', 'HeimVision Smart WiFi Camera', 'Wyze Camera', 'Amazon Echo Show'}
- count of different traffic or attack types
    - BN       35529
    - BF       20552
    - CI        6962
    - XSS       6612
    - BM        6138
    - Mirai     1115
- 76908 total points
